test/acceptance/features/bindWithCustomNamingStrategies.feature

```diff
@@ -0,0 +1,235 @@
+                name: backend-custom-file-naming
+                annotations:
+                    service.binding/host_cross_ns_service: path={.spec.host_cross_ns_service}
+            spec:
+                host_cross_ns_service: cross.ns.service.stable.example.com
+            """
+        * Generic test application "myapp-custom-file-naming" is running
+        When Service Binding is applied
+            """
+            apiVersion: operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: binding-request-custom-file-naming
+            spec:
+                namingStrategy: "PREFIX_{{ .service.kind | upper }}_{{ .name | upper }}_SUFFIX"
+                mountPath: "/foo/bar"
+                bindAsFiles: true
+                application:
+                    name: myapp-custom-file-naming
+                    group: apps
+                    version: v1
+                    resource: deployments
+                services:
+                -   group: stable.example.com
+                    version: v1
+                    kind: Backend
+                    name: backend-custom-file-naming
+            """
+        Then Service Binding "binding-request-custom-file-naming" is ready
+        And Content of file "/foo/bar/PREFIX_BACKEND_HOST_CROSS_NS_SERVICE_SUFFIX" in application pod is
+            """
+            cross.ns.service.stable.example.com
+            """
+
+    @error-naming
+    Scenario: Bind an application to a service with naming strategy error
+        * OLM Operator "backend" is running
+        * The Custom Resource is present
+            """
+            apiVersion: stable.example.com/v1
+            kind: Backend
+            metadata:
+                name: backend-naming-error
+                annotations:
+                    service.binding/host_cross_ns_service: path={.spec.host_cross_ns_service}
+            spec:
+                host_cross_ns_service: cross.ns.service.stable.example.com
+            """
+        * Generic test application "myapp-naming-error" is running
+        When Service Binding is applied
+            """
+            apiVersion: operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: binding-request-naming-error
+            spec:
+                namingStrategy: "{{ .service.test.name | lower }}_incorrect"
+                application:
+                    name: myapp-naming-error
+                    group: apps
+                    version: v1
+                    resource: deployments
+                services:
+                -   group: stable.example.com
+                    version: v1
+                    kind: Backend
+                    name: backend-naming-error
+            """
+        Then Service Binding "binding-request-naming-error" is ready
+        And The application env var "host_cross_ns_service" has value "cross.ns.service.stable.example.com"
```

Comment:
As a user, I would not expect that the binding is successful if I there is an error while generating binding names, quite contrary.

Reply:
done